File: AUGUST2020/2.c
typedef struct {
    bool* li;
} MyHashSet;

/** Initialize your data structure here. */

MyHashSet* myHashSetCreate() {
    MyHashSet *l=malloc(sizeof(MyHashSet));
    l->li=calloc(1000001,sizeof(bool));
    return l;
}

void myHashSetAdd(MyHashSet* obj, int key) {
  obj->li[key]=true;
}

void myHashSetRemove(MyHashSet* obj, int key) {
  obj->li[key]=false;
}

/** Returns true if this set contains the specified element */
bool myHashSetContains(MyHashSet* obj, int key) {
if (obj->li[key])
      return true;
  return false;
}

void myHashSetFree(MyHashSet* obj) {
    free(obj->li);
    free(obj);
}

/**
 * Your MyHashSet struct will be instantiated and called as such:
 * MyHashSet* obj = myHashSetCreate();
 * myHashSetAdd(obj, key);
 
 * myHashSetRemove(obj, key);
 
 * bool param_3 = myHashSetContains(obj, key);
 
 * myHashSetFree(obj);
*/

File: AUGUST2020/2.py
class MyHashSet:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.l=[0]*(1000001)
        

    def add(self, key: int) -> None:
        self.l[key]=1        

    def remove(self, key: int) -> None:
        self.l[key]=0

    def contains(self, key: int) -> bool:
        """
        Returns true if this set contains the specified element
        """
        if self.l[key]:
            return True
        return False
        


# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(key)
# obj.remove(key)
# param_3 = obj.contains(key)
